src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/RampedHaplotypeCallerEngine.java

```diff
@@ -0,0 +1,619 @@
+            return;
+        }
+
+        context.givenAlleles = context.features.getValues(hcArgs.alleles).stream()
+                .filter(vc -> hcArgs.forceCallFiltered || vc.isNotFiltered()).collect(Collectors.toList());
+
+        if (context.givenAlleles.isEmpty() && context.region.size() == 0) {
+            // No reads here so nothing to do!
+            context.regionVariants = referenceModelForNoVariation(context.region, true, context.VCpriors);
+        }
+    }
+
+    private void assemble(final CallRegionContext context) {
+
+        try {
+            // no need for this step?
+            if (context.regionVariants != null) {
+                return;
+            }
+
+            if (assemblyDebugOutStream != null) {
+                assemblyDebugOutStream.write("\n\n\n\n" + context.region.getSpan() + "\nNumber of reads in region: " + context.region.getReads().size() + "     they are:");
+                for (GATKRead read : context.region.getReads()) {
+                    assemblyDebugOutStream.write(read.getName() + "   " + read.convertToSAMRecord(context.region.getHeader()).getFlags());
+                }
+            }
+
+            // get off?
+            if (preAssemblerOffRamp != null) {
+                preAssemblerOffRamp.add(context.region, "assemblyResult", context.assemblyResult, context.nonVariantLeftFlankRegion, context.nonVariantRightFlankRegion, readsHeader);
+                context.regionVariants = NO_CALLS;
+                return;
+            }
+
+            // assembler on ramp?
+            CallRegionContext debugContext = null;
+            if (postAssemblerOnRamp != null) {
+
+                // clean part
+                context.nonVariantLeftFlankRegion = postAssemblerOnRamp.getOptionalAssemblyRegion(context.region, "assemblyResult.nonVariantLeftFlankRegion", readsHeader);
+                context.nonVariantRightFlankRegion = postAssemblerOnRamp.getOptionalAssemblyRegion(context.region, "assemblyResult.nonVariantRightFlankRegion", readsHeader);
+                context.assemblyResult = postAssemblerOnRamp.getAssembyResult(context.region, "assemblyResult", readsHeader, hcArgs, logger, aligner, referenceReader);
+                if (context.assemblyResult == null) {
+                    context.regionVariants = NO_CALLS;
+                    return;
+                }
+                context.haplotypeCollapsing = context.assemblyResult.getHaplotypeCollapsingEngine();
+                RampUtils.logReads(rpArgs.rampsDebugReads, "onramp: reads before trimming", context.assemblyResult.getRegionForGenotyping().getReads());
+
+                RampUtils.logReads(rpArgs.rampsDebugReads, "onramp: BEFORE untrimmedAssemblyResult reads", context.region.getReads());
+                final AssemblyResultSet untrimmedAssemblyResult = AssemblyBasedCallerUtils.assembleReads(context.region, context.givenAlleles, hcArgs, readsHeader, samplesList, logger, referenceReader, assemblyEngine, aligner,
```

Comment:
`context.givenAlleles` is outdated here. The forced alleles are handled later, this refers to the forced alleles that might have come from the pileup code.

Reply:
Resolved